### PR TITLE
Add SafeLogging to more classes

### PR DESCRIPTION
**Pull request by dvogelbacher**
## What changes were proposed in this pull request?
Add SafeLogging (similar to https://github.com/palantir/spark/pull/425) to more classes that can be useful:
- k8s classes
- CoarseGrainedSchedulerBackend
- SparkContext
- MemoryStore
- Executor
- CoarseGrainedExecutorBackend
- TorrentBroadCast and Broadcast

### CODE REVIEW

File: resource-managers/yarn/src/main/scala/org/apache/spark/scheduler/cluster/YarnSchedulerBackend.scala [hunk 1]
```diff
@@ -24,6 +24,7 @@ import scala.concurrent.ExecutionContext.Implicits.global
 import scala.util.{Failure, Success}
 import scala.util.control.NonFatal
 
+import com.palantir.logsafe.UnsafeArg
```
Comment (yifeih) [inline resource-managers/yarn/src/main/scala/org/apache/spark/scheduler/cluster/YarnSchedulerBackend.scala]: I think you have to import SafeLogging here?

File: FORK.md [hunk 2]
```diff
@@ -18,4 +18,5 @@
 # Added
 
 * Gradle plugin to easily create custom docker images for use with k8s
-* Filter rLibDir by exists so that daemon.R references the correct file [460](https://github.com/palantir/spark/pull/460) 
+* Filter rLibDir by exists so that daemon.R references the correct file [460](https://github.com/palantir/spark/pull/460)
+* SafeLogging
```
Comment (robert3005) [inline FORK.md]: this has to be more descriptive

File: resource-managers/yarn/src/main/scala/org/apache/spark/scheduler/cluster/YarnSchedulerBackend.scala (hunk 1 above)
Comment (robert3005) [reply resource-managers/yarn/src/main/scala/org/apache/spark/scheduler/cluster/YarnSchedulerBackend.scala]: comes transitively from CoarseGrainedSchedulerBackend

**Comment by robert3005** [comment]
LGTM but you should add more descriptive entry in fork - in particular I would like to see list of classes with modified logging

File: core/src/main/scala/org/apache/spark/broadcast/TorrentBroadcast.scala [hunk 3]
```diff
@@ -226,10 +230,13 @@ private[spark] class TorrentBroadcast[T: ClassTag](obj: T, id: Long)
...
           case None =>
-            logInfo("Started reading broadcast variable " + id)
+            safeLogInfo("Started reading broadcast variable",
+              SafeArg.of("id", id))
             val startTimeMs = System.currentTimeMillis()
             val blocks = readBlocks()
-            logInfo("Reading broadcast variable " + id + " took" + Utils.getUsedTimeMs(startTimeMs))
+            safeLogInfo("Reading broadcast variable finished",
+              SafeArg.of("id", id),
+              SafeArg.of("timeUsed", Utils.getUsedTimeMs(startTimeMs)))
```
[the remark this answers is not in the document]
Comment (dvogelbacher) [reply core/src/main/scala/org/apache/spark/broadcast/TorrentBroadcast.scala]: `Utils.getUsedTimeMs` already includes the unit https://github.com/palantir/spark/blob/master/core/src/main/scala/org/apache/spark/util/Utils.scala#L1051

File: core/src/main/scala/org/apache/spark/executor/Executor.scala [hunk 4]
```diff
@@ -59,9 +60,11 @@ private[spark] class Executor(
     userClassPath: Seq[URL] = Nil,
     isLocal: Boolean = false,
     uncaughtExceptionHandler: UncaughtExceptionHandler = new SparkUncaughtExceptionHandler)
-  extends Logging {
+  extends SafeLogging {
 
-  logInfo(s"Starting executor ID $executorId on host $executorHostname")
+  safeLogInfo("Starting executor on host",
+    SafeArg.of("executorId", executorId),
+    UnsafeArg.of("host", executorHostname))
```
[the remark this answers is not in the document]
Comment (dvogelbacher) [reply core/src/main/scala/org/apache/spark/executor/Executor.scala]: I think they should be, as they could be arbitrary?

File: FORK.md [hunk 5]
```diff
@@ -10,12 +10,15 @@
 * [SPARK-18079](https://issues.apache.org/jira/browse/SPARK-18079) - CollectLimitExec.executeToIterator should perform per-partition limits
 
 * [SPARK-15777](https://issues.apache.org/jira/browse/SPARK-15777) (Partial fix) - Catalog federation
- * make ExternalCatalog configurable beyond in memory and hive
- * FileIndex for catalog tables is provided by external catalog instead of using default impl
+    * make ExternalCatalog configurable beyond in memory and hive
```
Comment (dvogelbacher) [inline FORK.md]: the formatting here seemed to be off so I fixed it as well

**Comment by dvogelbacher** [comment]
@yifeih I think all these new classes can be quite useful for debugging a spark apps lifecycle and are needed in order for this safe logging to be effective. If you think a particular one might not be useful at all then I can take it out again. I also wouldn't expect too many merge conflicts as these are all core classes that don't tend to change much.

**Comment by robert3005** [comment]
👍

**Pull request merged**